File: SortingBasics/Insertion Sort/JS Implementation/InsertionSort.js
function sortList(unsortedList)
{

        for (let i = 0; i < unsortedList.length; i++)
        {
            let current = i;
            while (current > 0 && current > unsortedList[current - 1])
            {
                const temp = unsortedList[current];
                unsortedList[current] = unsortedList[current - 1];
                unsortedList[current - 1] = temp;
                current--;
            }
        }
        return unsortedList;
}

function splitWords(s) {
    return s == "" ? [] : s.split(' ');
}

function* main() {
    const unsortedList = splitWords(yield).map((v) => parseInt(v));
    const res = sortList(unsortedList);
    console.log(res.join(' '));
}

class EOFError extends Error {}
{
    const gen = main();
    const next = (line) => gen.next(line).done && process.exit();
    let buf = '';
    next();
    process.stdin.setEncoding('utf8');
    process.stdin.on('data', (data) => {
        const lines = (buf + data).split('\n');
        buf = lines.pop();
        lines.forEach(next);
    });
    process.stdin.on('end', () => {
        buf && next(buf);
        gen.throw(new EOFError());
    });
}

File: SortingBasics/Insertion Sort/InsertionSort.py
from typing import list
def sort_list(unsorted_list: List[int]) -> List[int]:
    for i, entry in enumerate(unsorted_list):
        current = i
        while current > 0 and unsorted_list[current] < unsorted_list[current - 1]:
            unsorted_list[current], unsorted_list[current-1] = unsorted_list[current-1], unsorted_list[current]
            current -= 1
    return unsorted_list

    if __name__ == '__main__':
    unsorted_list = [int(x) for x in input().split()]
    res = sort_list(unsorted_list)
    print(' '.join(map(str, res)))